atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ManyClientPoolingContainer.java

```diff
@@ -40,14 +40,18 @@
 
 public class ManyClientPoolingContainer extends ForwardingPoolingContainer<Client>
         implements ManyHostPoolingContainer<Client> {
+
     private static final Logger log = LoggerFactory.getLogger(ManyClientPoolingContainer.class);
-    volatile ImmutableList<PoolingContainer<Client>> containers = ImmutableList.of();
+
+    private volatile ImmutableList<PoolingContainer<Client>> containers = ImmutableList.of();
+
     @GuardedBy("this")
-    final Map<InetSocketAddress, PoolingContainer<Client>> containerMap = Maps.newHashMap();
+    private final Map<InetSocketAddress, PoolingContainer<Client>> containerMap = Maps.newHashMap();
+
     @GuardedBy("this")
-    boolean isShutdown = false;
+    private boolean isShutdown = false;
+
     boolean safetyDisabled = false;
-    private final Random random = new Random();
```

Comment:
Docs say it's thread safe, as of Java 1.4 http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6362070 ?